src/sentry/api/base.py

```diff
@@ -151,6 +157,10 @@ def dispatch(self, request, *args, **kwargs):
         self.request = request
         self.headers = self.default_response_headers  # deprecate?
 
+        # Tags that will ultimately flow into the metrics backend at the end of
+        # the request (happens via middleware/stats.py).
+        request._metric_tags = {}
```

Comment:
Do you need this if this property is conditionally set by `add_metric_tags`?

Reply:
I wanted to ensure it was always set, in case nothing in the request path calls `add_metrics_tags`. And I conditionally set it there just because of paranoia 😛